sdk/core/azure-core/src/main/java/com/azure/core/util/FluxUtil.java

```diff
@@ -270,20 +279,164 @@ public static boolean isFluxByteBuffer(Type entityType) {
+                return super.getData(null);
             }
 
-            if (!context.isEmpty()) {
-                context.stream().forEach(entry ->
-                    azureContext[0] = azureContext[0].addData(entry.getKey(), entry.getValue()));
+            // Look through the reactor context first
+            Optional<Object> value = reactorContext.getOrEmpty(key);
+            if (value.isPresent()) {
+                return value;
             }
 
-            return serviceCall.apply(azureContext[0]);
-        });
+            // then look through the map that we were given
+            if (contextAttributes != null && contextAttributes.containsKey(key)) {
+                return Optional.ofNullable(contextAttributes.get(key));
+            }
+
+            // otherwise defer up the context chain
+            return super.getData(key);
+        }
+
+        @Override
+        public Map<Object, Object> getValues() {
+            return new AbstractMap<Object, Object>() {
+
+                @Override
+                public int size() {
+                    return size;
+                }
+
+                @Override
+                public boolean isEmpty() {
+                    return size == 0;
+                }
+
+                @Override
+                public boolean containsKey(Object key) {
+                    return (contextAttributes != null && contextAttributes.containsKey(key))
+                               || reactorContext.hasKey(key);
+                }
+
+                @Override
+                public boolean containsValue(Object value) {
+                    throw LOGGER.logExceptionAsWarning(new UnsupportedOperationException());
+                }
+
+                @Override
+                public String get(final Object key) {
+                    if (reactorContext.hasKey(key)) {
+                        return reactorContext.get(key);
+                    } else if (contextAttributes != null && contextAttributes.containsKey(key)) {
+                        return contextAttributes.get(key);
+                    } else {
+                        return null;
+                    }
+                }
+
+                @Override
+                public String put(Object key, Object value) {
+                    throw LOGGER.logExceptionAsWarning(new UnsupportedOperationException());
+                }
+
+                @Override
+                public String remove(Object key) {
+                    throw LOGGER.logExceptionAsWarning(new UnsupportedOperationException());
+                }
+
+                @Override
+                public void putAll(Map<? extends Object, ? extends Object> m) {
+                    throw LOGGER.logExceptionAsWarning(new UnsupportedOperationException());
+                }
+
+                @Override
+                public void clear() {
+                    throw LOGGER.logExceptionAsWarning(new UnsupportedOperationException());
+                }
```

Comment:
These are runtime breaking changes as we previously returned a HashMap that allowed these operations.